src/dbstructureqitemviewfacade.h

```diff
@@ -0,0 +1,107 @@
+#ifndef DBSTRUCTUREQITEMVIEWFACADE_H
+#define DBSTRUCTUREQITEMVIEWFACADE_H
+
+#include <QObject>
+#include <QString>
+#include <QAbstractItemView>
+#include <QTreeView>
+#include <DbStructureModel.h>
+
+#include <sql/ObjectIdentifier.h>
+
+
+/**
+ * @brief The DbStructureQItemViewFacade class
+ * Provides a simple read only interface to a QAbstractItemView (e.g. QTreeView) that holds a model() of type DBStructureModel.
+ *
+ * It is designed to simplify the access to the actual selected node.
+ *
+ * The Class follows the facade design pattern.
+ * But it doesn't control the lifecycle of the itemView it is connected to.
+ *
+ */
+class DbStructureQItemViewFacade : public QObject
+{
+    Q_OBJECT
+public:
+    explicit DbStructureQItemViewFacade(QAbstractItemView *aitemView = new QTreeView());
```

Comment:
I don't understand why you are specifying a default value here. I thought there is no point creating a ```DbStructureQItemViewFacade``` object without a tree view object. But if there is no point in doing so, this should be made clear to the user of the class by not even allowing it :smile: